consensus-engine/src/lib.rs

```diff
@@ -31,7 +32,8 @@ impl<O: Overlay> Carnot<O> {
             highest_voted_view: View(-1),
             last_view_timeout_qc: None,
             overlay,
-            safe_blocks: [(genesis_block.id, genesis_block)].into(),
+            safe_blocks: [(genesis_block.id, genesis_block.clone())].into(),
+            tip: genesis_block,
```

Comment:
Nitpick: The tip could be just the `safe_blocks` index.

Reply:
That would actually be much nicer. I'll focus now on a testnet instance, do you mind taking this over?

Reply:
Sure! I'll do it!